### PR TITLE
Ik/enable checks

### PR DESCRIPTION
**Pull request by Ilya-Krylov**

### CODE REVIEW

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
@druzhkov-paul there are some pylint issues, will fix them later, failed checks (mypy, pylint, etc) does not cause failed build (temporarily)

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
@druzhkov-paul could you have a look, there are changes mostly made automatically by tools (flake8, black, isort) mostly

**Comment by samet-akcay** [comment]
Great to see that the repo is using isort black and pre-commit  👏. Would be super cool to have mypy too. Is it part of the plan @Ilya-Krylov ?

**Comment by Ilya-Krylov** [comment]
@samet-akcay  this PR adds mypy as well for OTE_SDK. Next [PR](https://github.com/openvinotoolkit/training_extensions/pull/706) will add these tools for other parts of code in the repo

**Comment by samet-akcay** [comment]
Does that mean that we'll be introducing type hinting throughout the repo to fully utilize mypy?

**Comment by Ilya-Krylov** [comment]
I guess we can. What do you think about it? Has it been already fully done for anomaly?

File: ote_cli/ote_cli/datasets/__init__.py [hunk 1]
```diff
@@ -1,35 +1,43 @@
...
 
 assert any((ObjectDetectionDataset, ClassificationDataset, SemanticSegmentationDataset))
 
 
 def get_dataset_class(name):
     registry = {
-        'detection': ObjectDetectionDataset,
-        'classification': ClassificationDataset,
-        'segmentation': SemanticSegmentationDataset
+        "detection": ObjectDetectionDataset,
```
Comment (druzhkov-paul) [inline ote_cli/ote_cli/datasets/__init__.py:30]: This is because SDK is set to use double quotes, right?

**Review by druzhkov-paul — APPROVED** [review]

**Comment by samet-akcay** [comment]
Yeah, `anomalib` is almost 100% annotated. There are couple of modules that we still need to annotate, but quite minor. we have a ticket for that.

I think it has several advantages. First of all, without type hints, mypy scan isn't fully utilised since mypy skips those that don't have type hints. Another advantage is the readability. With type-hints, the code becomes much more readable. I find this useful especially for the second advantage.

File: ote_cli/ote_cli/datasets/__init__.py (hunk 1 above)
Comment (Ilya-Krylov) [reply ote_cli/ote_cli/datasets/__init__.py:30]: right

**Pull request merged**